src/tests/test_prep.py

```diff
@@ -57,8 +60,8 @@ def test_prep_willingness():
     assert sum(pop.get_variable(col.PREP_CAB_WILLING, over_15s)) == 0
     assert sum(pop.get_variable(col.PREP_LEN_WILLING, over_15s)) == 0
     assert sum(pop.get_variable(col.PREP_VR_WILLING, over_15s)) == 0
-    # check for highest preference (ignore floats to discount nan values for under 15s)
-    assert all([i[0] is PrEPType.Oral for i in pop.data[col.PREP_PREF_RANKED] if type(i) is not float])
+    # check oral is highest preference (ignore unassigned 0 ranks for under 15s)
+    assert all(pop.data[col.PREP_ORAL_RANK] <= 1)
```

Comment:
Why not just check `==1` for over 15s?

Reply:
This would be the more precise way of going about things, I just initially wanted to do this check in one assert when I first made the rank changes. I'll split this up to check for == 0 for under 15s and == 1 for over 15s.